amethyst_renderer/src/pass/skybox/interleaved.rs

```diff
@@ -0,0 +1,118 @@
+//! Skybox pass
+
+use amethyst_core::{
+    cgmath::{Matrix4, One},
+    specs::{Read, ReadStorage},
+    transform::GlobalTransform,
+};
+use gfx::pso::buffer::ElemStride;
+use glsl_layout::{mat4, Uniform};
+use std::marker::PhantomData;
+use {
+    error::Result,
+    get_camera,
+    pass::util::set_attribute_buffers,
+    pipe::{
+        pass::{Pass, PassData},
+        DepthMode, Effect, NewEffect,
+    },
+    set_vertex_args, ActiveCamera, Camera, Encoder, Factory, Mesh, Normal, PosNormTex, Position,
+    Query, Shape, TexCoord, VertexFormat,
+};
+
+use super::{SkyboxColor, FRAG_SRC, VERT_SRC};
+
+#[repr(C, align(16))]
+#[derive(Clone, Copy, Debug, Uniform)]
+pub(crate) struct VertexArgs {
+    proj: mat4,
+    view: mat4,
+    model: mat4,
+}
+
+/// Draw a simple gradient skybox
+///
+/// # Type Parameters:
+///
+/// * `V`: `VertexFormat`
+#[derive(Clone, Debug)]
+pub struct DrawSkybox<V> {
+    _marker: PhantomData<V>,
+    mesh: Option<Mesh>,
+}
+
+impl<V> DrawSkybox<V>
+where
+    V: Query<(Position, Normal, TexCoord)>,
+{
+    /// Create instance of `DrawSkybox` pass
+    pub fn new() -> Self {
+        DrawSkybox {
+            _marker: PhantomData {},
+            mesh: None,
+        }
+    }
+}
+
+impl<'a, V> PassData<'a> for DrawSkybox<V>
+where
+    V: Query<(Position, Normal, TexCoord)>,
+{
+    type Data = (
+        Option<Read<'a, ActiveCamera>>,
+        ReadStorage<'a, Camera>,
+        ReadStorage<'a, GlobalTransform>,
+        Read<'a, SkyboxColor>,
+    );
+}
+
+impl<V> Pass for DrawSkybox<V>
+where
+    V: Query<(Position, Normal, TexCoord)>,
+{
+    fn compile(&mut self, mut effect: NewEffect) -> Result<Effect> {
+        let verts = Shape::Cube.generate_vertices::<Vec<PosNormTex>>(None);
+        self.mesh = Some(Mesh::build(verts).build(&mut effect.factory)?);
+
+        effect
+            .simple(VERT_SRC, FRAG_SRC)
+            .without_back_face_culling()
```

Comment:
You can probably enable culling, no?

Reply:
This was the only function I could see that enabled back faces. Is there another way to configure culling modes?

Reply:
Ah okay so the mesh should really be generated differently. I'll probably clean this up later so this is fine for now :smile: